packages/core/admin/server/src/services/homepage.ts

```diff
@@ -0,0 +1,131 @@
+          action: 'plugin::content-manager.explorer.read',
+        },
+      });
+      const allowedContentTypeNames = readPermissions
+        .map((permission) => permission.subject)
+        .filter(Boolean) as UID.ContentType[];
+
+      // Fetch the configuration for each content type in a single query
+      const rawConfigurations = await coreStore.findMany({
+        where: {
+          key: {
+            $in: allowedContentTypeNames.map(
+              (contentType) => `plugin_content_manager_configuration_content_types::${contentType}`
+            ),
+          },
+        },
+      });
+      const configurations = rawConfigurations.map((rawConfiguration) => {
+        return JSON.parse(rawConfiguration.value);
+      });
+
+      const recentDocuments = await Promise.all(
+        allowedContentTypeNames.map(async (contentTypeName) => {
+          const configuration = configurations.find((config) => config.uid === contentTypeName);
+          const contentType = strapi.contentType(contentTypeName);
+          const fields = ['documentId', 'updatedAt'];
+
+          // Add fields required to get the status if D&P is enabled
+          const hasDraftAndPublish = contentTypes.hasDraftAndPublish(contentType);
+          if (hasDraftAndPublish) {
+            fields.push('publishedAt');
+          }
+
+          // Only add the main field if it's defined
+          const { mainField } = configuration.settings;
+          if (mainField) {
+            fields.push(mainField);
+          }
+
+          // Only add locale if it's localized
+          const isLocalized = (contentType.pluginOptions?.i18n as any)?.localized;
+          if (isLocalized) {
+            fields.push('locale');
+          }
+
+          const documents = await strapi.documents(contentTypeName).findMany({
+            limit: MAX_DOCUMENTS,
+            sort: 'updatedAt:desc',
+            fields,
+          });
+
+          return documents.map((document) => {
+            /**
+             * Save the main field value before deleting it so we can use the common
+             * title key instead across all content types. Use the delete operator instead of
+             * destructuring or lodash omit for better type inference.
+             */
+            const mainFieldValue = document[mainField ?? 'documentId'];
+            delete document[mainField];
+
+            return {
+              data: {
+                ...document,
+                updatedAt: new Date(document.updatedAt),
+                title: mainFieldValue,
+              },
+              meta: {
+                model: contentTypeName,
+                kind: contentType.kind,
+                hasDraftAndPublish,
+              },
+            };
+          });
+        })
+      );
+
+      const overallRecentDocuments = recentDocuments
+        .flat()
+        .sort((a, b) => b.data.updatedAt.valueOf() - a.data.updatedAt.valueOf())
+        .slice(0, MAX_DOCUMENTS);
```

Comment:
Shouldn't you slice first to only sort on the length of MAX_DOCUMENTS

Reply:
I don't think I can because before the sort all the documents are one content type after the other. So if I slice before I sort then I'll only keep content type A docs even if content type Z docs are more recent